crates/fluvio/src/config/cluster.rs

```diff
@@ -58,6 +62,74 @@ impl FluvioConfig {
         self.tls = tls.into();
         self
     }
+
+    pub fn query_metadata_path<'de, T>(&self, path: &str) -> Option<T>
+    where
```

Comment:
Don't see use case for supporting nesting (not sure even that make sense).  In fact it would make supporting complex.  Just need to way to support multiple annotations.   With strong typing, any nesting would be taken care bout annotation structure

Reply:
okay, when we discussed the feature we also talked about supporting nesting, but I can remove it, no problem.

Reply:
Nesting is not needed since you should be able to map arbitrary struct which can be any nested

Reply:
so, do you mean we should remove support for nested queries?

`query_metadata_path("arbitrary.struct.with.nested.levels")`

if so, then comes the problem pointed out in the previous PR: it's less convenient because in order to update/query a specific value in some arbitrary level of metadata you need to duplicate code or add the crate that defines the type as a dependency to the crate manipulating the type.

Reply:
Yes.  Can you give example why nesting queries are necessary?  

For example, let's say you have following annotation:
```
struct MyAnnotation {
     name: String,
     size: u64,
     field1: Field1,
     field2: Field2,
     items: Vec<Field2>
}

struct Field1 {
     width: u32,
     length: u16
}

struct Field2 {
      speed: u64
}
```

So any kind of nesting can be accomplished with this scheme

Reply:
maybe we're only interested in `Field1.width` and don't want to deserialize everything else? or maybe `MyAnnotation` is defined at crate `A`, and we want to query `Field2.speed` in crate `B` but don't want to add crate `A` as a dependency?

I'm basing these assumptions on this comment: https://github.com/infinyon/fluvio/pull/3628#issuecomment-1788534087

Reply:
They can always parse `Value` if don't want to add dependency but this will leads other issue due to lack of strong typing.  This is limitation with Annotation.   Its ok to push burden to Annotation consumer/producer.